examples/AuctionGen.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE UndecidableInstances #-}
+
+module AuctionGen where
+
+import Hedgehog.Gen (
+  choice,
+  integral,
+ )
+
+import Hedgehog.Plutus.Gen
+
+import AuctionExample (
+  Auction (Auction),
+  AuctionDatum (AuctionDatum),
+  AuctionTest (AuctionTest),
+  AuctionTestRedeemer (TestRedeemerBid),
+  Bid (Bid),
+  SelfOutput (SelfOutput),
+ )
+
+import Hedgehog.Plutus.TestData (Good, TestData (generalise))
+import Hedgehog.Range qualified as Range
+
+genGoodAuctionTest :: GenContext (Good AuctionTest)
+genGoodAuctionTest =
```

Comment:
The idea was to construct a `Generics.SOP` type directly. That way, the data is guaranteed correct by construction. I agree this is very lacking ergonomically; I will have to come up with a solution.